src/RouteCollection.php

```diff
@@ -55,171 +72,174 @@ public function __construct(
+            if (! is_null($route->getScheme()) && $route->getScheme() !== $request->getUri()->getScheme()) {
+                continue;
+            }
+
+            // check for domain condition
+            if (! is_null($route->getHost()) && $route->getHost() !== $request->getUri()->getHost()) {
+                continue;
+            }
+
+            $route->setContainer($this->container);
+
+            if (is_null($route->getStrategy())) {
+                $route->setStrategy($this->getStrategy());
+            }
+
+            $this->addRoute(
+                $route->getMethods(),
+                $this->parseRoutePath($route->getPath()),
+                [$route, 'dispatch']
+            );
+        }
     }
 
     /**
-     * Add a route that responds to DELETE HTTP method
+     * Build an index of named routes.
      *
-     * @param  string                                   $route
-     * @param  string|\Closure                          $handler
-     * @param  \League\Route\Strategy\StrategyInterface $strategy
-     * @return \League\Route\RouteCollection
+     * @return void
      */
-    public function delete($route, $handler, Strategy\StrategyInterface $strategy = null)
+    protected function buildNameIndex()
     {
-        return $this->addRoute('DELETE', $route, $handler, $strategy);
+        $this->processGroups();
+
+        foreach ($this->routes as $key => $route) {
+            if (! is_null($route->getName())) {
+                unset($this->routes[$key]);
+                $this->namedRoutes[$route->getName()] = $route;
+            }
+        }
     }
 
     /**
-     * Add a route that responds to HEAD HTTP method
+     * Process all groups.
      *
-     * @param  string                                   $route
-     * @param  string|\Closure                          $handler
-     * @param  \League\Route\Strategy\StrategyInterface $strategy
-     * @return \League\Route\RouteCollection
+     * @return void
      */
-    public function head($route, $handler, Strategy\StrategyInterface $strategy = null)
+    protected function processGroups()
     {
-        return $this->addRoute('HEAD', $route, $handler, $strategy);
+        foreach ($this->groups as $key => $group) {
+            unset($this->groups[$key]);
+            $group();
+        }
     }
 
-    /**
-     * Add a route that responds to OPTIONS HTTP method
-     *
-     * @param  string                                   $route
-     * @param  string|\Closure                          $handler
-     * @param  \League\Route\Strategy\StrategyInterface $strategy
-     * @return \League\Route\RouteCollection
-     */
-    public function options($route, $handler, Strategy\StrategyInterface $strategy = null)
+    public function getNamedRoute($name)
     {
-        return $this->addRoute('OPTIONS', $route, $handler, $strategy);
+        $this->buildNameIndex();
+
+        if (array_key_exists($name, $this->namedRoutes)) {
```

Comment:
Except that `isset` and `array_key_exists` have slightly meaning, specifically around `null` values. There is no way to be *absolutely* sure that a `null` value will not get into `namedRoutes`, especially since it is only `protected`.